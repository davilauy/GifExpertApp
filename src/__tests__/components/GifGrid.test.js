import "@testing-library/jest-dom";
import React from "react";

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";

import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Prueba en: <GifGrid />", () => {
  const category = "Will Smith";

  test("Debe de mostrar el componente correctamente ", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar items cuando se cargan imagenes desde custom hook ", () => {
    const dataMock = [
      {
        id: "abc",
        title: "test",
        url: "https://localhost/img.png",
      },
      {
        id: "123",
        title: "test 2",
        url: "https://localhost/img.png",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: dataMock,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(dataMock.length);
  });
});
