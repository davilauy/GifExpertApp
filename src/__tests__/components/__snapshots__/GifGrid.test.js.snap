// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en: <GifGrid /> Debe de mostrar el componente correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Will Smith
  </h3>
  <p
    className="card-grid animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`Prueba en: <GifGrid /> Debe de mostrar items cuando se cargan imagenes desde custom hook  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Will Smith
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="test"
      url="https://localhost/img.png"
    />
    <GifGridItem
      id="123"
      key="123"
      title="test 2"
      url="https://localhost/img.png"
    />
  </div>
</Fragment>
`;
